compiler/ballerina-lang/src/main/java/io/ballerina/projects/DiagnosticResult.java

```diff
@@ -32,13 +32,18 @@
     protected Collection<Diagnostic> errors;
     protected Collection<Diagnostic> warnings;
     protected Collection<Diagnostic> hints;
+    protected Collection<Diagnostic> internalExcluded;
 
     protected DiagnosticResult(Collection<Diagnostic> allDiagnostics) {
         this.allDiagnostics = Collections.unmodifiableCollection(allDiagnostics);
     }
 
     public Collection<Diagnostic> diagnostics() {
-        return allDiagnostics;
+        return diagnostics(true);
+    }
+
+    public Collection<Diagnostic> diagnostics(boolean includeInternal) {
+        return includeInternal ? allDiagnostics : getInternalExcluded(allDiagnostics);
     }
```

Comment:
@nadeeshaan By looking at the existing APIs in this class, seems like they have prevented being too generic and have tried to be specific everywhere. I think this is to avoid unintended usages. That's why I did not expose a generic one using a predicate. WDYT?

Reply:
Yeah, that seems so. Then that's fine to proceed